plugins/catalog-react/src/components/EntityTables/SystemsTable.tsx

```diff
@@ -0,0 +1,157 @@
+/*
+ * Copyright 2020 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  EntityName,
+  RELATION_OWNED_BY,
+  RELATION_PART_OF,
+  SystemEntity,
+} from '@backstage/catalog-model';
+import { Table, TableColumn } from '@backstage/core';
+import { makeStyles } from '@material-ui/core';
+import React from 'react';
+import { getEntityRelations } from '../../utils';
+import {
+  EntityRefLink,
+  EntityRefLinks,
+  formatEntityRefTitle,
+} from '../EntityRefLink';
+
+type EntityRow = {
+  entity: SystemEntity;
+  resolved: {
+    name: string;
+    partOfDomainRelationTitle?: string;
+    partOfDomainRelations: EntityName[];
+    ownedByRelationsTitle?: string;
+    ownedByRelations: EntityName[];
+  };
+};
+
+const columns: TableColumn<EntityRow>[] = [
+  {
+    title: 'Name',
+    field: 'resolved.name',
+    highlight: true,
+    render: ({ entity }) => (
+      <EntityRefLink entityRef={entity} defaultKind="System" />
+    ),
+  },
+  {
+    title: 'Domain',
+    field: 'resolved.partOfDomainRelationTitle',
+    render: ({ resolved }) => (
+      <EntityRefLinks
+        entityRefs={resolved.partOfDomainRelations}
+        defaultKind="domain"
+      />
+    ),
+  },
+  {
+    title: 'Owner',
+    field: 'resolved.ownedByRelationsTitle',
+    render: ({ resolved }) => (
+      <EntityRefLinks
+        entityRefs={resolved.ownedByRelations}
+        defaultKind="group"
+      />
+    ),
+  },
+  {
+    title: 'Description',
+    field: 'entity.metadata.description',
+    width: 'auto',
+  },
+];
+
+type Props = {
+  title: string;
+  variant?: string;
+  entities: SystemEntity[];
+  emptyComponent?: JSX.Element;
+};
+
+const useStyles = makeStyles(theme => ({
+  empty: {
+    padding: theme.spacing(2),
+    display: 'flex',
+    justifyContent: 'center',
+  },
+}));
+
+export const SystemsTable = ({
```

Comment:
Tbh I think `SystemsTable` (and component) are too specific for this package. Perhaps move to the catalog plugin for now or smth like that?

I don't think this package should contain any logic related to specific entity kinds, since that's a dimension that is completely open to customization, while this package isn't. Perhaps the best solution here would be a slightly lower level `EntityTable`?

Reply:
On the other hand, does it still make sense to have things here that pertain to the _core_ kinds? It's a pretty well defined small set of kinds and semantics after all. I would like to look at this package as "the toolbox of pieces with which you can glue together your own entity page etc"

Reply:
As a possible way to quickly make everyone happy, is a single `EntityTable` a reasonable refactor? or does it reduce the value of the component too much

Reply:
A `EntityTable` would be possible, but the question is, what columns we have in such a table. As far as I see, the only thing all entities have in common are name/entity ref & description (from the parts the tables display right now). So I'm not sure.

Reply:
So maybe to be clear why export these from `catalog-react` at all and not just having the entity cards: I reuse the `ComponentsTable` in `api-docs`. We could move both to `catalog` and not export them and duplicate the `ComponentsTable` into `api-docs`.

Reply:
Hmm, what do you think about an `EntityTable` + `EntityTable.componentColumns` + `EntityTable.systemColumns` combo? Along with moving over the entity field picking into the column definitions

Provides a clear structure and I think would be simple enough to use, but also space for completely custom solutions.

Something like

```tsx
<EntitiesTable
  title="Consumers"
  variant={variant}
  emptyComponent={ /* bleh */ }
  entities={entities}
  columns={EntitiesTable.componentColumns}
/>
```

Not sure if the columns would be found at `EntitiesTable.*` or not, but tbh they have to exist somewhere, so 😅

Reply:
I will have a look tomorrow. Just wondering where to do the transformation from entity to row. To make the table work properly I transform the values of the relations before passing them into the table. But these transforms are entity kind specific. However, I may saw a different solution some days ago...

Reply:
@backstage/maintainers on board with ^ btw? Before @Fox32 puts in work towards that 😅

Reply:
😆 already playing with it

Reply:
`columns: 'domain-defaults' | 'system-defaults' | string[]`? :)

Reply:
Definitions are a lot more complex than that tho

Reply:
And we agreed that letting it live in the react package was good, right

Reply:
Ya right, so something like `() => ColumnWhateverType[]` then (that only gets evaluated on first render) perhaps, but the idea is the same

Reply:
What do you think about something like this? https://github.com/SDA-SE/backstage/commit/42df711a93e1959d9004442ab88f57d10e41e47f

Reply:
@Fox32 👌